compiler/ir/backend.jvm/entrypoint/src/org/jetbrains/kotlin/backend/jvm/JvmIrCodegenFactory.kt

```diff
@@ -171,8 +171,12 @@ open class JvmIrCodegenFactory(
         val result = LinkedHashSet<ModuleDescriptor>()
         fun collectImpl(descriptor: ModuleDescriptor) {
             val dependencies = descriptor.allDependencyModules
-            dependencies.forEach { if (it !in result) collectImpl(it) }
-            result += dependencies
+            dependencies.forEach {
+                if (it !in result) {
```

Comment:
Very minor, but I suppose you could say `if (!result.add(it)) collectImpl(it)`

Reply:
Yes! Thank you. :)

Reply:
And restored it as a one-liner.